### PR TITLE
Add `X2Y2: Exchange`

### PR DESCRIPTION
**Pull request by shermiexyz**
Add `X2Y2: Exchange`

### CODE REVIEW

**Comment by aminhadson** [comment]
why? is it dydx?!

File: adapters/x2y2/index.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
+require("dotenv").config();
+
+const moment = require("moment");
+const BigNumber = require("bignumber.js");
```
Comment (zqx210) [inline adapters/x2y2/index.js:4]: fdfds

**Review by zqx210 — APPROVED** [review]

File: adapters/x2y2/abi.json [hunk 2]
```diff
@@ -0,0 +1,1086 @@
+[
+  {
```
Comment (zqx210) [inline adapters/x2y2/abi.json:2]: 12fsoia

Comment (zqx210) [reply adapters/x2y2/abi.json:2]: f kauuw

**Review by zqx210 — APPROVED** [review]
kaisk-x2y2

**Comment by zqx210** [comment]
dd a

**Comment by shijinlong558** [comment]
nice

File: adapters/x2y2/index.js (hunk 1 above)
Comment (shijinlong558) [reply adapters/x2y2/index.js:4]: gh pr checkout 17

**Comment by liutaurasjanuskas** [comment]
Already integrated by DappRadar team: [https://dappradar.com/nft/marketplaces/protocol/ethereum](https://dappradar.com/nft/marketplaces/protocol/ethereum)

**Pull request closed without merging**